src/synapse/azext_synapse/_params.py

```diff
@@ -0,0 +1,161 @@
+            c.argument('tags', arg_type=tags_type, help='The tags of spark batch job.')
+            c.argument('detailed', arg_type=get_three_state_flag(),
+                       help='Optional query parameter specifying whether detailed response is returned beyond plain livy.')
+
+        with self.argument_context(scope + ' list') as c:
+            c.argument('from_index', help='Optional parameter specifying which index the list should begin from.')
+            c.argument('detailed', arg_type=get_three_state_flag(),
+                       help='Optional query parameter specifying whether detailed response is returned beyond plain livy.')
+            c.argument('size',
+                       help='The size of the returned list.By default it is 20 and that is the maximum.')
+
+    with self.argument_context('synapse spark batch create') as c:
+        c.argument('job_type', arg_type=get_enum_type(['SCALA', 'PYTHON', 'DOTNET'], default='SCALA'),
+                   help='The spark batch job type.')
+
+    for scope in ['show', 'cancel']:
+        with self.argument_context('synapse spark batch ' + scope) as c:
+            c.argument('batch_id', options_list=['--id'], arg_group='Spark Batch',
+                       help='The id of the spark batch job.')
+
+    with self.argument_context('synapse spark batch show') as c:
+        c.argument('detailed', arg_type=get_three_state_flag(),
+                   help='Optional query parameter specifying whether detailed response is returned beyond plain livy.')
+
+    for scope in ['show', 'cancel', 'reset-timeout']:
+        with self.argument_context('synapse spark session ' + scope) as c:
+            c.argument('session_id', options_list=['--id'], arg_group='Spark Session',
+                       help='The id of the spark session job.')
+
+    with self.argument_context('synapse spark session show') as c:
+        c.argument('detailed', arg_type=get_three_state_flag(),
+                   help='Optional query parameter specifying whether detailed response is returned beyond plain livy.')
+
+    with self.argument_context('synapse spark session-statement') as c:
+        c.argument('session_id', help='The id of spark session job.')
+
+    for scope in ['show', 'cancel']:
+        with self.argument_context('synapse spark session-statement ' + scope) as c:
+            c.argument('statement_id', options_list=['--id'], arg_group="Spark Session-statement",
+                       help='The id of the statement.')
+
+    with self.argument_context('synapse spark session-statement create') as c:
+        c.argument('code', help='The code of spark statement.')
+        c.argument('kind', help='The kind of spark statement.')
+
+    # synapse workspace
+    for scope in ['synapse workspace', 'synapse spark pool', 'synapse sql pool']:
+        with self.argument_context(scope) as c:
+            c.argument('resource_group_name', help='The resource group name.')

```

Comment:
use `arg_type=resource_group_name_type`. Another choice is just to omit this parameter here. CLI core handles it automatically for you when the parameter name is `resource_group_name` in custom.py methods.

Reply:
> use `arg_type=resource_group_name_type`. Another choice is just to omit this parameter here. CLI core handles it automatically for you when the parameter name is `resource_group_name` in custom.py methods.

fixed